packages/app-shell-shared/src/utils/processConfig.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * Process raw config to add $key properties
+ * This enables React key-based reconciliation
+ */
+
+import type {
+  HvAppShellConfig,
+  HvAppShellMenuConfig,
+  HvAppShellViewsConfig,
+} from "../types/Config";
+
+/**
+ * Process config to add $key properties
+ * Returns processed config for React key-based reconciliation
+ */
+export const processConfig = (
+  rawConfig: HvAppShellConfig,
+): HvAppShellConfig => {
+  let keyCounter = 0;
+
+  // Helper to add key (if not present)
+  const registerElement = <T extends { $key?: string | number }>(
+    element: T,
+  ): T => {
+    return element.$key == null ? { ...element, $key: keyCounter++ } : element;
```

Comment:
IMO, we should actually overwrite any `$key` value the user might have specified, making sure they are all unique.